charms/worker/k8s/src/utils.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utility functions."""
+
+import re
+
+# Constants for time calculations
+YEAR_SECONDS = 365 * 24 * 60 * 60  # seconds in a year
+MONTH_SECONDS = 30 * 24 * 60 * 60  # seconds in a month
+DAY_SECONDS = 24 * 60 * 60  # seconds in a day
+
+
+def ttl_to_seconds(ttl: str):
```

Comment:
Does this match some kind of well known format?  Something in my head is going off because this module doesn't include `datetime` or `timedelta` at all.  Help me down from the cliff here.  

I found a few packages that do this kind of datetime math:

https://arrow.readthedocs.io/en/latest/index.html#
https://bear.im/code/parsedatetime/docs/index.html

Reply:
Yes, it's the agreed format in the snap (and pretty much in other parts of Kubernetes upstream). For instance: https://github.com/canonical/k8s-snap/blob/6ee37dbc0124c5099338daae8ced9af24689db7d/src/k8s/pkg/utils/time.go#L43